guppy/views.py

```diff
@@ -7,3 +11,22 @@ class TermsOfServiceView(TemplateView):
 
 class PrivacyPolicyView(TemplateView):
     template_name = "guppy/privacy_policy.html"
+
+
+class ReferralProgramView(TemplateView):
+    template_name = "guppy/referral_program.html"
+
+    def get_context_data(self, **kwargs):
+        # TODO: pass referral info to the view
+        context = super().get_context_data(**kwargs)
+        # Pass referral link
+        referral_link, created = ReferralLink.objects.get_or_create(
```

Comment:
@autn should this be created here? Shouldn't this be created when the user is created, not when they access this view?

Reply:
@YPCrumble we can move this to create user method, but for existing users, we will create manually?

Reply:
@YPCrumble I tried to add a `save()` method in the model, but the `ReferralLink` requires `UserProfile`, which is not created, so generating the link on the first time getting profile is fine. Does that make sense?

Reply:
@autn totally fine to create manually for existing users, and also (because we haven't gotten any users yet!) it's totally fine to delete all uses on the production server.

On the save method - once you call `super().save()` inside that method, shouldn't you then have a `UserProfile` that's saved to the db?

Reply:
@autn also this isn't a huge deal - if it's easier just add a `TODO` saying "move this to the model save method in the future" or something like that :D.

Reply:
@YPCrumble thank you, I got it. 

> On the save method - once you call super().save() inside that method, shouldn't you then have a UserProfile that's saved to the db?
Why didn't I think of this? :smile: